velox/dwio/orc/test/ReaderTest.cpp

```diff
@@ -277,3 +287,2869 @@ TEST_F(OrcReaderTest, testOrcRlev2) {
     EXPECT_EQ(nameCol->valueAt(0), "AAAA");
   }
 }
+class OrcFileDescription {
+ public:
+  std::string filename;
+  std::string json;
+  std::string typeString;
+  std::string formatVersion;
+  std::string softwareVersion;
+  uint64_t rowCount;
+  uint64_t contentLength;
+  uint64_t stripeCount;
+  common::CompressionKind compression;
+  size_t compressionSize;
+  uint64_t rowIndexStride;
+  std::map<std::string, std::string> userMeta;
+
+  OrcFileDescription(
+      const std::string& _filename,
+      const std::string& _json,
+      const std::string& _typeString,
+      const std::string& _version,
+      const std::string& _softwareVersion,
+      uint64_t _rowCount,
+      uint64_t _contentLength,
+      uint64_t _stripeCount,
+      common::CompressionKind _compression,
+      size_t _compressionSize,
+      uint64_t _rowIndexStride,
+      const std::map<std::string, std::string>& _meta)
+      : filename(_filename),
+        json(_json),
+        typeString(_typeString),
+        formatVersion(_version),
+        softwareVersion(_softwareVersion),
+        rowCount(_rowCount),
+        contentLength(_contentLength),
+        stripeCount(_stripeCount),
+        compression(_compression),
+        compressionSize(_compressionSize),
+        rowIndexStride(_rowIndexStride),
+        userMeta(_meta) {}
+
+  friend std::ostream& operator<<(
+      std::ostream& stream,
+      OrcFileDescription const& obj);
+};
+
+std::ostream& operator<<(std::ostream& stream, OrcFileDescription const& obj) {
+  stream << obj.filename;
+  return stream;
+}
+
+std::map<std::string, std::string> makeMetadata();
```

Comment:
@kevins963 Is this used anywhere? The implementation is a huge > 2000 line blob.